packages/lib-runtime/src/store/coderefs.ts

```diff
@@ -165,8 +185,15 @@ export const resolveCodeRefValues = async <TExtension extends Extension>(extensi
   await Promise.allSettled(resolutions);
 
   if (resolutionErrors.length > 0) {
-    throw new CodeRefError('Failed to resolve code references', resolutionErrors);
+    throw new ExtensionCodeRefsResolutionError(
+      'Failed to resolve code references',
+      resolutionErrors,
+      extension,
+    );
   }
 
-  return extension as ResolvedExtension<TExtension>;
+  return {
+    ...extension,
```

Comment:
This preserves referential identity of all direct `extension` properties (and consequently their nested properties), except for `properties` property which we clone manually.

Cloning `extension.properties` should be perfectly safe:
- raw extensions JSON data arrives (as part of the plugin manifest)
- raw extensions JSON data is parsed and validated via `loader.getPluginManifest`
- extensions have their `EncodedCodeRef` values decoded into `CodeRef` functions via `decodeCodeRefs`
- plugin containing these extensions is added to `PluginStore`

Once added to the `PluginStore`, extensions should not be modified directly. Their properties are either static data or `CodeRef` functions.